pkg/kubelet/cm/dra/state/state_checkpoint.go

```diff
@@ -55,35 +49,25 @@ type ClaimInfoState struct {
 	// PodUIDs is a set of pod UIDs that reference a resource
 	PodUIDs sets.Set[string]
 
-	// CDIDevices is a map of DriverName --> CDI devices returned by the
-	// GRPC API call NodePrepareResource
-	CDIDevices map[string][]string
+	// DriverState contains information about all drivers which have allocation
+	// results in the claim, even if they don't provide devices for their results.
+	DriverState map[string]DriverState
 }
 
-// ClaimInfoStateWithoutResourceHandles is an old implementation of the ClaimInfoState
-// TODO: remove in Beta
-type ClaimInfoStateWithoutResourceHandles struct {
-	// Name of the DRA driver
-	DriverName string
-
-	// ClassName is a resource class of the claim
-	ClassName string
-
-	// ClaimUID is an UID of the resource claim
-	ClaimUID types.UID
-
-	// ClaimName is a name of the resource claim
-	ClaimName string
-
-	// Namespace is a claim namespace
-	Namespace string
-
-	// PodUIDs is a set of pod UIDs that reference a resource
-	PodUIDs sets.Set[string]
+// DriverState is used to store per-device claim info state in a checkpoint
+// +k8s:deepcopy-gen=true
+type DriverState struct {
+	Devices []Device
+}
 
-	// CDIDevices is a map of DriverName --> CDI devices returned by the
-	// GRPC API call NodePrepareResource
-	CDIDevices map[string][]string
+// Device is how a DRA driver described an allocated device in a claim
+// to kubelet. RequestName and CDI device IDs are optional.
+// +k8s:deepcopy-gen=true
+type Device struct {
```

Comment:
This is the same struct as in the api.proto. I defined it again because I didn't want to make the checkpoint depend directly on an API definition that could change - that is what caused https://github.com/kubernetes/kubernetes/issues/123552.

We still need versioning to handle the situation that we need to extend the checkpoint content.